gwos-evm/polyjuice-tests/fuzz/gw-syscall-simulator/src/lib.rs

```diff
@@ -0,0 +1,714 @@
+            .map_err(|err| anyhow!("syscall error: get script hash by account id : {:?}", err))?;
+        Ok(value)
+    }
+
+    fn get_script(&mut self, script_hash: &H256) -> Option<Script> {
+        self.state.get_script(script_hash)
+    }
+    fn get_data(&self, data_hash: &H256) -> Option<Bytes> {
+        self.state.get_data(data_hash)
+    }
+}
+
+static HOST: Lazy<Mutex<GodwokenHost>> = Lazy::new(|| Mutex::new(GodwokenHost::new()));
+
+#[no_mangle]
+pub unsafe extern "C" fn ckb_exit(code: i8) -> i32 {
+    std::process::exit(code.into());
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn ckb_debug(s: *const c_char) {
+    let message = CStr::from_ptr(s).to_str().expect("UTF8 error!");
+    println!("Debug message: {}", message);
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn gw_load_rollup_config(addr: *mut c_void, len: *mut u64) -> c_int {
+    let data = HOST.lock().unwrap().rollup_config.as_slice().to_vec();
+    store_data(addr, len, 0, &data);
+    SUCCESS
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn gw_store(key_addr: *const u8, value_addr: *const u8) -> c_int {
+    let key = load_data_h256(key_addr);
+    let value = load_data_h256(value_addr);
+    HOST.lock()
+        .unwrap()
+        .state
+        .update_raw(key, value)
+        .expect("gw_store");
+    SUCCESS
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn gw_load(key_addr: *const u8, value_addr: *mut u8) -> c_int {
+    let key = load_data_h256(key_addr);
+    let val = HOST.lock().unwrap().state.get_raw(&key).expect("gw_load");
+    store_h256(value_addr, &val);
+    SUCCESS
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn gw_set_return_data(addr: *const u8, len: u64) -> c_int {
+    let buf = load_bytes(addr, len);
+    HOST.lock().unwrap().run_result.return_data = buf;
+    SUCCESS
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn gw_create(
+    script_addr: *const u8,
+    script_len: u64,
+    account_id_addr: *mut u32,
+) -> c_int {
+    let script_data = load_bytes(script_addr, script_len);
+    let script = match Script::from_slice(&script_data[..]) {
+        Ok(script) => script,
+        Err(_err) => return GW_ERROR_INVALID_ACCOUNT_SCRIPT.into(),
+    };
+    let script_hash = script.hash();
+    let mut host = HOST.lock().unwrap();
+    // check exists
+    match host.get_account_id_by_script_hash(&script_hash.into()) {
+        Ok(Some(_)) => return GW_ERROR_DUPLICATED_SCRIPT_HASH.into(),
+        Ok(None) => {}
+        Err(_err) => {
+            return ERROR;
+        }
+    }
+    // TODO: valide script
+    let id = match host.get_account_count() {
+        Ok(id) => id,
+        Err(_err) => {
+            return ERROR;
+        }
+    };
+
+    let account_nonce_key = build_account_field_key(id, GW_ACCOUNT_NONCE_TYPE);
+    host.state
+        .update_raw(account_nonce_key, H256::zero())
+        .expect("account nonce key");
+    host.run_result.write_data_hashes.insert(account_nonce_key);
```

Comment:
https://github.com/godwokenrises/godwoken/blob/develop/crates/store/src/state/state_db.rs#L499

The `write_data_hashes` and `read_data_hashes` are only related to these syscalls https://github.com/godwokenrises/godwoken/blob/develop/crates/generator/src/syscalls/mod.rs#L396-L428 we shouldn't insert state key into the `data_hashes`.

Please fix it.

Reply:
Fixed. Those data hashes were stored in the wrong place. And run_result cannot be covered in the fuzz testing, so we can just delete them safely.